File: web-content/src/app/pod/members/model/PersonalInformation.ts
import {Interest} from '../edit/PersonalInformationEditor.component';

export class PersonalInformation {
    interests: Interest[] = [];
    email: string = '';
    firstName: string = '';
    lastName: string = '';
    phoneNumber: string = '';


    constructor(personalInformation?: any) {
        if(personalInformation){
            this.interests = personalInformation.interests.map((it: Interest) => new Interest(it.id, it.value));
            this.email = personalInformation.email;
            this.firstName = personalInformation.firstName;
            this.lastName = personalInformation.lastName;
            this.phoneNumber = personalInformation.phoneNumber;
        }

    }

    addInterest(interest: Interest): void {
        this.interests.push(interest)
    }

    removeInterest(interest: Interest): void {
        this.interests = this.interests.filter(it => it.id !== interest.id)
    }
}

File: web-service/src/main/java/io/acari/event/source/models/PersonalInformation.java
package io.acari.event.source.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class PersonalInformation implements Contactable {
    private List<Interest> interests = new ArrayList<>();
    private String email;
    private String firstName;
    private String lastName;
    private String phoneNumber;
}
